opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -1004,11 +1005,17 @@ export const hashMergeValidation = (instances) => {
 // region mutation update
 const ed = (date) => isEmptyField(date) || date === FROM_START_STR || date === UNTIL_END_STR;
 const noDate = (e) => ed(e.first_seen) && ed(e.last_seen) && ed(e.start_time) && ed(e.stop_time);
-const filterTargetByExisting = (targetEntity, redirectSide, sourcesDependencies, targetDependencies) => {
-  const filtered = [];
+const filterTargetByExisting = async (context, targetEntity, redirectSide, sourcesDependencies, targetDependencies) => {
   const cache = [];
+  const filtered = [];
   const sources = sourcesDependencies[`i_relations_${redirectSide}`];
   const targets = targetDependencies[`i_relations_${redirectSide}`];
+  const markingSources = sources.filter((r) => r.i_relation.entity_type === RELATION_OBJECT_MARKING);
```

Comment:
Can you give more information about this changes ?
Thank you :)

Reply:
Yes, more details can be provided by @richard-julien :)

Reply:
It resolve the problem of duplication when merging entities.
Some markings relations need to be removed, some need to be kept and some needs to be moved, depending if the direction.